LuckParser/Models/BossLogic/Cairn.cs

```diff
@@ -47,10 +47,26 @@ protected override CombatReplayMap GetCombatMapInternal()
         
         public override void ComputeAdditionalBossData(Boss boss, ParsedLog log)
         {
-            // TODO: needs doughnuts (wave) and facing information (sword)
+            // TODO: needs doughnuts (wave)
+            CombatReplay replay = boss.CombatReplay;
+            List<CastLog> cls = boss.GetCastLogs(log, 0, log.FightData.FightDuration);
             switch (boss.ID)
             {
                 case (ushort)ParseEnum.BossIDS.Cairn:
+                    List<CastLog> swordSweep = cls.Where(x => x.SkillId == 37631).ToList();
+                    foreach (CastLog c in swordSweep)
+                    {
+                        int start = (int)c.Time;
+                        int preCastTime = 1400;
+                        int initialHitDuration = 850;
+                        int sweepDuration = 1100;
+                        int width = 1400; int height = 80;
+                        Point3D facing = replay.Rotations.FirstOrDefault(x => x.Time >= start);
```

Comment:
you need to check that facing != null, this will crash on logs without rotation tracking